torch/fx/experimental/proxy_tensor.py

```diff
@@ -458,6 +458,9 @@ def wrap_key(f, tensors, tracer):
     @functools.wraps(f)
     def wrapped(*proxies):
         flat_proxies, proxies_spec = pytree.tree_flatten(proxies)
+        if len(flat_proxies) == len(flat_tensors) + 1:
+            assert isinstance(flat_proxies[0], torch.nn.Module)
+            flat_proxies = flat_proxies[1:]
```

Comment:
I think I see what's going on here, but this still deserves a comment.

Reply:
Comments added.

Reply:
I'm missing the context of why we're calling `make_fx` on modules directly?

Reply:
Why can't we do this by lifting the parameters to inputs and adjusting the names that way?

Reply:
I agree that it's also good to do it in param lifting manner. It's just current implementation that dynamo.export calls make_fx directly on a graph module, which didn't do the graph_lifting.

I think the pr may still be valid in the following sense: 1. the main part of this pr tries to improve Tracer.trace(): current implementation doesn't preserve the module attr information when the module to be traced is wrapped by functions. This modification could be meaningful for fx tracer in general. 2. It also provides some convenience when we are able to call make_fx on a graph module. Users (e.g. me) may have the wrong impression that make_fx should also work for nn.Module. 

Maybe an alternative could be that we make our intention more explicit: instead of inline the logic for discarding nn.Module in wrap_key, we could create a separate wrap_mod_key function, which checks whether it's a wrapper of a module and redirects wrap_key.

Or we just state clearly that calling make_fx on module is prohibited. And to workaround, users could do a param lifting and unlifting after the make_fx returns. Both approaches are OK to me XD.

Reply:
What is the result of calling `make_fx` on a `nn.Module` after this PR? IMO, lifting the parameters explicitly is my preferred way of doing it, for a number of reasons related to things like fake tensors or symbolic shapes.

If the issue is that even after lifting your params the names are terrible, I think it wouldn't be a bad idea to add some utility/way inside of `make_fx` of providing names to your inputs other than implicitly pulling it from the name of the arguments to your function (so usually `params`).

Reply:
I've updated the before and after comparison for dynamo.export and make_fx in the summary. I can understand the fake tensors and symbolics shapes concerns. If we create a separate wrap_mod_key function, can we do the lifting/un-lifting and tensor tracking there?

Reply:
I would prefer to simply `make_fx` through calling the module with `stateless.functional_call`. Similar to how we do it in `aot_autograd.py`.

Reply:
Since export would also like to run graph with interpreter for propagating the stack_trace, it seems that we don't want to use the API directly (or not?).  I checked aot_module_simplified and find we need some of work (potentially redundant) to achieve the expected result (i.e. the calling convention and module attr access path etc are preserved after dynamo.export() so that the downstream EXIR stack works correctly).

I have the feeling that the choice here is that:
1. Either **we don't want to support make_fx for nn.Module at all** (even in the future) and we expect users to do the param/buffer lifting thing before make_fx. But this may require some redundant effort across the code base to tailor their specific needs. 
2. **we'd like to support make_fx for nn.Module someday.** In this case, we need to define a contract of the returned graph_module. This pr can be taken as an initial attempt to do that (i.e. correctly preserving the wrapped module information in the tracer). The next step could be param lifting or tracking parameters and buffers within make_fx. 

Any comments? @ezyang